examples/structured_data/classification_with_tfdf.py

```diff
@@ -383,22 +387,22 @@ def call(self, inputs):
         # Cast the inputs int64 a tensor.
         inputs = tf.cast(inputs, tf.dtypes.int64)
         # Lookup positive frequencies for the input feature values.
-        positive_fequency = tf.cast(
+        positive_frequency = tf.cast(
             tf.gather_nd(self.positive_frequency_lookup, inputs),
             dtype=tf.dtypes.float32,
         )
         # Lookup negative frequencies for the input feature values.
-        negative_fequency = tf.cast(
+        negative_frequency = tf.cast(
             tf.gather_nd(self.negative_frequency_lookup, inputs),
             dtype=tf.dtypes.float32,
         )
         # Compute positive probability for the input feature values.
-        positive_probability = positive_fequency / (
-            positive_fequency + negative_fequency
+        positive_probability = positive_frequency / (
+            positive_frequency + negative_frequency
         )
         # Concatenate and return the looked-up statistics.
         return tf.concat(
-            [positive_fequency, negative_fequency, positive_probability], axis=1
+            [positive_frequency, negative_frequency, positive_probability], axis=1
         )
 
```

Comment:
Copy from https://github.com/keras-team/keras-io/pull/782

Could the colab report the test accuracy of the three tested methods?

Example: https://www.tensorflow.org/decision_forests/tutorials/model_composition_colab#whats_next

I don't see the output of the colab in this review.


>> TensorFlow Decision Forests provide powerful models, especially with structured data.
In our experiments, the Gradient Boosted Tree model achieved 95.79% test accuracy.
When using the target encoding with categorical feature, the same model achieved 95.81% test accuracy.
When pretraining embeddings to be used as inputs to the Gradient Boosted Tree model,
we achieved 95.82% test accuracy.

If the original model has a better accuracy than the model with pre-processing, this might indicate a bug in the code. I would not leave it this way. Could it be the smoothing hyper-parameter?

Alternatively, I would separate the simple GBT code from the experimental pre-processing in two separate examples (files). Users that just want to learn about GBT would benefit from it.